lib/ansible/modules/crypto/get_certificate.py

```diff
@@ -119,8 +131,9 @@
 from ansible.module_utils.basic import AnsibleModule, missing_required_lib
 
 from os.path import isfile
-from ssl import get_server_certificate
-from socket import setdefaulttimeout
+from ssl import get_server_certificate, create_default_context, DER_cert_to_PEM_cert, CERT_NONE, CERT_OPTIONAL
```

Comment:
```suggestion
from ssl import get_server_certificate, DER_cert_to_PEM_cert, CERT_NONE, CERT_OPTIONAL
import traceback

CREATE_DEFAULT_CONTEXT_IMP_ERR = None
try:
  from ssl import create_default_context
except ImportError as dummy:
  CREATE_DEFAULT_CONTEXT_IMP_ERR = traceback.format_exc()
  HAS_CREATE_DEFAULT_CONTEXT = False
else:
  HAS_CREATE_DEFAULT_CONTEXT = True

```
Then you can fail the module if `proxy_host` is specified and `HAS_CREATE_DEFAULT_CONTEXT` is `False`.

You can fail somehow like this:
```.py
module.fail_json(msg='To use proxy_host, you must run the get_certificate module with Python 2.7 or newer.'),
                                     exception=CREATE_DEFAULT_CONTEXT_IMP_ERR)
```

Reply:
Thank you for teaching me!
Add a `create_default_context` module import check processing.

Reply:
Rethinking this, you could probably also use something like `if (sys.version_info.major, (sys.version_info.minor) >= (2, 7): from ssl import create_default_context` to make sure this is only imported for Python 2.7 or newer. But the import guard also has its advantages (in case that function is removed in a later Python version).